velocity/src/main/java/me/lorenzo0111/rocketjoin/listener/JoinListener.java

```diff
@@ -74,16 +85,26 @@ public void onJoin(PostLoginEvent e) {
             serializationException.printStackTrace();
         }
 
-        if (condition == null) {
+        if (condition == null && e.getPlayer().getCurrentServer().isPresent()) {
             boolean join = plugin.getConfig().join().enabled();
             Component message = plugin.parse(plugin.getConfig().join().message(),p);
+            Component otherServerMessage =
+                    plugin.parse(plugin.getConfig().join().otherServerMessage()
+                            .replace("{server}", p.getCurrentServer().get().getServerInfo().getName()), p);
+            ArrayList<RegisteredServer> otherServers = new ArrayList<>(plugin.getServer().getAllServers());
+            otherServers.remove(p.getCurrentServer().get().getServer());
             if (join) {
                 plugin.getServer().getScheduler().buildTask(plugin, () -> {
-                    for (Audience audience : plugin.getServer().getAllPlayers()) {
-                        audience.sendMessage(message);
+                    for (Audience audience : p.getCurrentServer().get().getServer().getPlayersConnected()) {
+                        if (e.getPreviousServer() != null) return;
```

Comment:
You shouldn’t use return, do yo mean continue?

Reply:
well if there is a previousServer then it means its a serverSwitch that should be called and not join.
Return basically just means it doesnt send 1 join and 1 serverswitch. You essentially dont want to continue the joinEvent if you know its a serverSwitch right?